sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2299,7 +2299,7 @@ class Analyzer(
         case u @ UpCast(child, _, _) if !child.resolved => u
 
         case UpCast(child, dataType, walkedTypePath)
-          if Cast.mayTruncate(child.dataType, dataType) =>
+          if !Cast.canSafeCast(child.dataType, dataType) =>
           fail(child, dataType, walkedTypePath)
```

Comment:
Is the error message of thrown `AnalysisException` valid under this change? Seems it is not just because truncating.

Reply:
updated